feat(cli): introduce `next info` CLI command (#32972)
vercel/next.js: docs/api-reference/cli.md

Participation in this anonymous program is optional, and you may opt-out if you'd not like to share any information.

To learn more about Telemetry, [please read this document](https://nextjs.org/telemetry/).

## Info

> Reporting accurate details about the environment in which Next.js runs when reporting a bug makes resolving the issue easier.

Reviewer: ```suggestion
`next info` prints details about the current system and environment.
```

> Useful information is then printed to the terminal, such as Operating System platform/version, Binaries (Node.js, npm, Yarn, pnpm) and npm package versions (`next`, `react`, `react-dom`). This eliminates the user having to call individual commands to gather the same information before opening a report.

Reviewer: ```suggestion
This information includes Operating System platform/arch/version, Binaries (Node.js, npm, Yarn, pnpm) and npm package versions (`next`, `react`, `react-dom`).
```

Running the following in your project's root directory:

```bash
next info
```

will give you information like this example:

```bash

    Operating System:

>       Platform: linux

Reviewer: ```suggestion
      Platform: linux
      Arch: x64
```

      Version: #22-Ubuntu SMP Fri Nov 5 13:21:36 UTC 2021
    Binaries:
      Node: 16.13.0
      npm: 8.1.0
      Yarn: 1.22.17
      pnpm: 6.24.2
    npm packages:

>       next: 12.0.8-canary.14

Reviewer: ```suggestion
      next: 12.0.8
```

      react: 17.0.2
      react-dom: 17.0.2

```

> This information should then be pasted into the bug report.

Reviewer: ```suggestion
This information should then be pasted into GitHub Issues.
```
